docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
+Kubernetes code itself.  Clients should consume the versioned APIs exclusively.
+
+What this means for API changes is that you have to be somewhat thoughtful in
+how you approach changes, and that you have to touch a number of pieces to make
+a complete change.  This document aims to guide you through the process - not
+all API changes will need all of these steps.
+
+## On compatibility
+
+Before talking about API changes, it is worthwhile to clarify what we mean by
+API compatibility.  An API change is considered compatible if it:
+   * adds new functionality that is not required for correct behavior
```

Comment:
It's worth calling out enums. We mostly avoid enums, but have a few de-facto ones. Perhaps counter-intuitively, *adding* a new enum value is a breaking change if clients might reasonably expect to understand all values a field may be set to. OTOH, *removing* an enum value, especially in a status field, can be a backward-compatible change, so long as it's treated similarly to a deprecated field -- the system doesn't reject the request or do something else unreasonable.